src/Akeneo/Category/back/Application/Enrichment/DeactivatedTemplateAttributesInValueCollectionCleaner.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Category\Application\Enrichment;
+
+use Akeneo\Category\Domain\Query\DeactivatedTemplateAttributes\DeactivatedTemplateAttributeIdentifier;
+use Akeneo\Category\Domain\ValueObject\Attribute\Value\AbstractValue;
+
+/**
+ * @copyright 2023 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class DeactivatedTemplateAttributesInValueCollectionCleaner
+{
+    /**
+     * @param array<DeactivatedTemplateAttributeIdentifier> $deactivatedAttributes
+     * @param array<string, string|null> $rawCategory
+     *
+     * @return array<string, string|null>
+     *
+     * @throws \JsonException
+     */
+    public function __invoke(array $deactivatedAttributes, array $rawCategory): array
+    {
+        if (empty($deactivatedAttributes) || empty($rawCategory['value_collection'])) {
+            return $rawCategory;
+        }
+        $decodedRawValueCollection = json_decode(
+            $rawCategory['value_collection'],
+            true,
+            512,
+            JSON_THROW_ON_ERROR,
+        );
+        foreach ($deactivatedAttributes as $deactivatedAttribute) {
+            $attributeCode = $deactivatedAttribute->code.AbstractValue::SEPARATOR.$deactivatedAttribute->uuid;
+            foreach ($decodedRawValueCollection as $key => $rawValue) {
+                if ($rawValue['attribute_code'] === $attributeCode) {
+                    unset($decodedRawValueCollection[$key]);
+                }
+            }
+        }
```

Comment:
We could avoid double foreach loop by using array_map and in_array functions.

```suggestion
        $deactivatedAttributeCodeList = array_map(fn ($attribute) => $attribute->code.AbstractValue::SEPARATOR.$attribute->uuid, $deactivatedAttributes);

        foreach ($decodedRawValueCollection as $key => $rawValue) {
            if (in_array($rawValue['attribute_code'], deactivatedAttributeCodeList) {
                unset($decodedRawValueCollection[$key]);
            }
        }
```